numba/cuda/tests/doc_examples/test_sessionize.py

```diff
@@ -0,0 +1,126 @@
+import unittest
+
+from numba.cuda.testing import CUDATestCase, skip_on_cudasim
+from numba.tests.support import captured_stdout
+
+
+@skip_on_cudasim("cudasim doesn't support cuda import at non-top-level")
+class TestSessionization(CUDATestCase):
+    """
+    Test click stream sessionization
+    """
+
+    def setUp(self):
+        # Prevent output from this test showing up when running the test suite
+        self._captured_stdout = captured_stdout()
+        self._captured_stdout.__enter__()
+        super().setUp()
+
+    def tearDown(self):
+        # No exception type, value, or traceback
+        self._captured_stdout.__exit__(None, None, None)
+        super().tearDown()
+
+    def test_ex_sessionize(self):
+        # ex_sessionize.import.begin
+        import numpy as np
+        from numba import cuda
+
+        # Set the timeout to one hour
+        session_timeout = np.int64(np.timedelta64("3600", "s"))
+        # ex_sessionize.import.end
+
+        # ex_sessionize.allocate.begin
+        # Generate data
+        ids = cuda.to_device(
+            np.array(
+                [
+                    1, 1, 1, 1, 1, 1,
+                    2, 2, 2,
+                    3, 3, 3, 3, 3, 3, 3, 3, 3, 3,
+                    4, 4, 4, 4, 4, 4, 4, 4, 4,
+                ]
+            )
+        )
+        sec = cuda.to_device(
+            np.array(
+                [
+                    1, 2, 3, 5000, 5001, 5002, 1,
+                    2, 3, 1, 2, 5000, 5001, 10000,
+                    10001, 10002, 10003, 15000, 150001,
+                    1, 5000, 50001, 15000, 20000,
+                    25000, 25001, 25002, 25003,
+                ],
+                dtype="datetime64[ns]",
+            ).astype(
+                "int64"
+            )  # Cast to int64 for compatibility
+        )
+        # Create a vector to hold the results
+        results = cuda.to_device(np.zeros(len(ids)))
+        # ex_sessionize.allocate.end
+
+        # ex_sessionize.kernel.begin
+        @cuda.jit
+        def sessionize(user_id, timestamp, results):
+            gid = cuda.grid(1)
+            size = len(user_id)
+
+            if gid >= size:
+                return
+
+            # Determine session boundaries
+            is_first_datapoint = gid == 0
+            if not is_first_datapoint:
+                new_user = user_id[gid] != user_id[gid - 1]
+                timed_out = (
+                    timestamp[gid] - timestamp[gid - 1] > session_timeout
+                )
+                is_sess_boundary = new_user or timed_out
+            else:
+                is_sess_boundary = True
+
+            # Determine session labels
+            if gid < size:
```

Comment:
I think this must always be true because of the `if gid >= size: return` above - can this conditional be removed?